pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -447,7 +447,21 @@ func (r *ReconcileBareMetalHost) actionDeleting(prov provisioner.Provisioner, in
 		info.log.Info("ready to be deleted")
 		return deleteComplete{}
 	}
-
+	// Try to power off host before delete except when it is externally provisioned.
+	if !info.host.Spec.ExternallyProvisioned {
+		provResult, err := prov.PowerOff()
```

Comment:
Do we need to check to see if the host is already powered off by looking at our status field?

Reply:
Done.

Reply:
Done.